app/Libraries/BeatmapsetDiscussionPostsBundle.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace App\Libraries;
+
+use App\Models\BeatmapDiscussionPost;
+use App\Traits\Memoizes;
+use App\Transformers\BeatmapDiscussionPostTransformer;
+use App\Transformers\BeatmapsetCompactTransformer;
+use App\Transformers\UserCompactTransformer;
+use Illuminate\Pagination\Paginator;
+
+class BeatmapsetDiscussionPostsBundle extends BeatmapsetDiscussionsBundleBase
+{
+    use Memoizes;
+
+    public function getData()
+    {
+        return $this->getPosts();
+    }
+
+    public function toArray()
+    {
+        return [
+            'beatmapsets' => json_collection($this->getBeatmapsets(), new BeatmapsetCompactTransformer()),
+            'cursor' => $this->getCursor(),
+            'posts' => json_collection($this->getPosts(), new BeatmapDiscussionPostTransformer()),
+            'users' => json_collection($this->getUsers(), new UserCompactTransformer()),
+        ];
+    }
+
+    private function getBeatmapsets()
+    {
+        return $this->memoize(__FUNCTION__, function () {
+            return $this->getPosts()->pluck('beatmapDiscussion.beatmapset')->uniqueStrict('id')->values();
```

Comment:
Preload for this seems missing. Or maybe it should be `->pluck('beatmapset')` instead?

Also there's nothing linking post and beatmapset with current json structure :upside_down_face: